Mathlib/Analysis/SpecialFunctions/Complex/Log.lean

```diff
@@ -243,3 +243,61 @@ theorem _root_.Continuous.clog {f : α → ℂ} (h₁ : Continuous f)
   continuous_iff_continuousAt.2 fun x => h₁.continuousAt.clog (h₂ x)
 
 end LogDeriv
+
+section tsum_tprod
+
+variable {α ι: Type*}
+
+open Real
+
+lemma Real.HasSum_rexp_HasProd  (f : ι → α → ℝ) (hfn : ∀ x n, 0 < f n x)
+    (hf : ∀ x : α, HasSum (fun n => log ((f n x))) (∑' i, log (f i x))) (a : α) :
+       HasProd (fun b ↦ f b a) ((rexp ∘ fun a ↦ ∑' (n : ι), log (f n a)) a) := by
```

Comment:
I can see why you wrote the `HasProd` as `rexp ∘ ...` but I think it makes more sense here to go straight to `Π`, no? I know the proof of some of these results may get *slightly* more complicated as a result, but I think it's worth it for a better statement. Same for the `Complex` version below.
```suggestion
    (hf : ∀ x : α, HasSum (fun n => log (f n x)) (∑' i, log (f i x))) (a : α) :
       HasProd (fun b ↦ f b a) ((rexp ∘ fun a ↦ ∑' (n : ι), log (f n a)) a) := by
```

Reply:
Do you mean just prove: `HasProd (fun b ↦ f b a) ∏' n : ι, ((f n a))`?

Reply:
Yep

Reply:
Ok cool, I don't know if I did it the most efficent way, but its not that much worse than what we had before.